### ISSUE TITLE
Regression: AKS cluster being recreated

### ISSUE THREAD

**Issue by yarinm** [pulumi/pulumi-azure-native#1532]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

We've been using v1.17.0 for quite some time, using go and the automation API

Our stack creates an AKS cluster with a default agent pool and we create an additional two  AgentPools using the `containerservice.NewAgentPool` API in the same stack.

We tried upgrading to v1.59.0 and we noticed the stack is forcibly recreating the AKS cluster. And even after the AKS cluster is recreated subsequent calls to stack update will cause the cluster to recreate.

In all these recreations the logs state that diff is:
```
aks updated [diff: ~agentPoolProfiles]
```
The issue also happens when I run the stack from scratch (meaning without any existing resources), subsequent updates still cause the AKS to recreate each time.

Another error I see occasionally during these cluster recreations:
```
azure-native:containerservice/v20210501:ManagedCluster (aks):
    error: 1 error occurred:
        * Code="ReconcileMSICredentialError" Message="Reconcile MSI credential failed. Details: autorest/azure: Service returned an error. Status=404 Code=\"NotFound\" Message=\"Resource '/subscriptions/XXXX/resourcegroups/dev-eu1-eastus2-xxx-orchestrator-rg-kMr7/providers/Microsoft.ContainerService/managedClusters/xxx-orchestrator-aks-v2p-eastus2-kmr7' was not found.\"." 
 ```
 
### Steps to reproduce
1. Create a stack that creates an AKS cluster with one agent pool, in the same stack create additional agent pools using `NewAgentPool` API. 
2. pulumi stack update
3. Change via API one of the agent pool's node count
4. pulumi stack update

Actual: AKS cluster is being recreated
Expected: Agent pool needs to surgically update and not force an entire cluster recreation

Note:
The issue might happen in older versions as well but since each test takes ~30 minutes due to the time it takes to create a cluster + testing if it is being recreated, I've only focused on the latest version.

**Comment by yarinm** [comment]
Further investigating this I think `ReconcileMSICredentialError` is probably related somehow to the version, the AKS is failing to recreate and is in a failed state.

I checked the change log in azure (some preview feature) about the AKS cluster and it says that these values are being removed:
```
properties.agentPoolProfiles["control1"].osSKU=Ubuntu => empty value
properties.agentPoolProfiles["control1"].enableFIPS=false => empty value
```

**Comment by viveklak** [comment]
@yarinm thanks for opening the issue and investigating with the change log feature!
Could you confirm that you are creating the same managed cluster at the same API version (v20210501)? To be precise, creating `azure-native:containerservice/v20210501:ManagedCluster` with 1.17.0 and adding additional agent pools results in the expected behavior. But the same code against 1.59.0 results in replacement?

**Comment by yarinm** [comment]
Correct, exact same code behaves differently in 1.59.0 and causing us to recreate the AKS cluster over and over again

**Comment by viveklak** [comment]
So I tried this @yarinm but I am seeing the same behavior with 1.17 and 1.59.0. Based on the information you shared, it seems you have a system agent pool created as part of the cluster and a subsequent separate nodepool, see the code I have been using below. Changing the node count for `pool1` doesn't seem to trigger the recreation. Could you provide some code to repro please?

```
                // Create the Azure Kubernetes Service cluster.
		cluster, err := containerservice.NewManagedCluster(ctx, "go-aks", &containerservice.ManagedClusterArgs{
			ResourceName:      pulumi.StringPtr("goakscluster"),
			ResourceGroupName: resourceGroup.Name,
			AgentPoolProfiles: containerservice.ManagedClusterAgentPoolProfileArray{
				&containerservice.ManagedClusterAgentPoolProfileArgs{
					Name:         pulumi.String("agentpool"),
					Mode:         pulumi.String("System"),
					OsDiskSizeGB: pulumi.Int(30),
					Count:        pulumi.Int(3),
					VmSize:       pulumi.String("Standard_DS2_v2"),
					OsType:       pulumi.String("Linux"),
				},
			},
			LinuxProfile: &containerservice.ContainerServiceLinuxProfileArgs{
				AdminUsername: pulumi.String("testuser"),
				Ssh: containerservice.ContainerServiceSshConfigurationArgs{
					PublicKeys: containerservice.ContainerServiceSshPublicKeyArray{
						containerservice.ContainerServiceSshPublicKeyArgs{
							KeyData: sshKey.PublicKeyOpenssh,
						},
					},
				},
			},
			DnsPrefix: resourceGroup.Name,
			ServicePrincipalProfile: &containerservice.ManagedClusterServicePrincipalProfileArgs{
				ClientId: adApp.ApplicationId,
				Secret:   adSpPassword.Value,
			},
			KubernetesVersion: pulumi.String("1.22.4"),
		})
		if err != nil {
			return err
		}

		nodepool, err := containerservice.NewAgentPool(ctx, "pool", &containerservice.AgentPoolArgs{
			AgentPoolName:     pulumi.String("pool1"),
			Count:             pulumi.Int(2),
			ResourceGroupName: resourceGroup.Name,
			ResourceName:      cluster.Name,
			VmSize:            pulumi.String("Standard_DS2_v2"),
			OsType:            pulumi.String("Linux"),
		}, pulumi.DependsOn([]pulumi.Resource{cluster}))
		if err != nil {
			return err
		}
```

**Comment by yarinm** [comment]
Vivek, AKS is sorting the pools by name. Try calling the system pool "control1" and the subsequent pool "analysis1".
I'll try to create a reproducable example in the next few days.

**Comment by yarinm** [comment]
[Stack that reproduces the issue](https://gist.github.com/yarinm/a0f69f48fe3d28d5559ffd5f1536f997)

The flow is

```bash
pulumi up
# change analysis1 agent pool count to zero (via CLI or portal)

# this will cause recreation of the cluster
pulumi up --refresh
```

Example run:

```bash
❯ pulumi up
Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE to remember):
Previewing update (dev):
     Type                                                       Name              Plan
 +   pulumi:pulumi:Stack                                        azure-go-aks-dev  create
 +   ├─ azure-native:resources/v20210401:ResourceGroup          azure-go-aks      create
 +   ├─ azure-native:containerservice/v20210501:ManagedCluster  aks               create
 +   └─ azure-native:containerservice/v20210501:AgentPool       analysis1         create

Resources:
    + 4 to create

Do you want to perform this update? yes
Updating (dev):
     Type                                                       Name              Status
 +   pulumi:pulumi:Stack                                        azure-go-aks-dev  created
 +   ├─ azure-native:resources/v20210401:ResourceGroup          azure-go-aks      created
 +   ├─ azure-native:containerservice/v20210501:ManagedCluster  aks               created
 +   └─ azure-native:containerservice/v20210501:AgentPool       analysis1         created

Resources:
    + 4 created

Duration: 8m37s
```

```bash
❯ PULUMI_CONFIG_PASSPHRASE=1234 pulumi up --refresh --yes
Previewing update (dev):
     Type                                                       Name              Plan        Info
     pulumi:pulumi:Stack                                        azure-go-aks-dev
     ├─ azure-native:resources/v20210401:ResourceGroup          azure-go-aks
 +-  ├─ azure-native:containerservice/v20210501:AgentPool       analysis1         replace     [diff:
 +-  └─ azure-native:containerservice/v20210501:ManagedCluster  aks               replace     [diff:

Resources:
    +-2 to replace
    2 unchanged

Updating (dev):
     Type                                                       Name              Status
     pulumi:pulumi:Stack                                        azure-go-aks-dev  running
     ├─ azure-native:resources/v20210401:ResourceGroup          azure-go-aks
 --  ├─ azure-native:containerservice/v20210501:AgentPool       analysis1         deleted original
 --  └─ azure-native:containerservice/v20210501:ManagedCluster  aks               deleting original
```

**Comment by viveklak** [comment]
So this is unfortunately expected behavior when mixing explicit agent pool subresources with managed clusters, see #579 for another instance of this. This is another instance of #1112. 

You are seeing the replace behavior after the update to the newer version due to #1210 which landed in v1.38.0. The best option for now is to consider transitioning the system agent pool to a separate AgentPool as an import after the initial cluster is created. 

I am going to mark this a duplicate of #1112.
